### PR TITLE
2022

### PR DESCRIPTION
**Pull request by laviRZ**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,83 +1,88 @@
...
 }
 
+def deployArtifact = deploy.targets.roborio.artifacts.frcJava
+
+// Set to true to use debug for JNI.
+wpi.java.debugJni = false
+
 // Set this to true to enable desktop support.
-def includeDesktopSupport = true
+def includeDesktopSupport = false
```
Comment (Mboesn) [inline build.gradle]: did you do this?

File: build.gradle [hunk 2]
```diff
@@ -1,83 +1,88 @@
...
+    nativeDebug wpi.java.deps.wpilibJniDebug(wpi.platforms.desktop)
+    nativeDebug wpi.java.vendor.jniDebug(wpi.platforms.desktop)
+    simulationDebug wpi.sim.enableDebug()
 
-    // Websocket extensions require additional configuration.
-    // simulation wpi.deps.sim.ws_server(wpi.platforms.desktop, false)
-    // simulation wpi.deps.sim.ws_client(wpi.platforms.desktop, false)
+    nativeRelease wpi.java.deps.wpilibJniRelease(wpi.platforms.desktop)
+    nativeRelease wpi.java.vendor.jniRelease(wpi.platforms.desktop)
+    simulationRelease wpi.sim.enableRelease()
```
Comment (Mboesn) [inline build.gradle:66]: did you do this?

File: src/main/java/frc/robot/RobotContainer.java [hunk 3]
```diff
@@ -39,11 +40,22 @@ private void initializeSubsystems() {
...
+        Supplier<Double> x, y, rot;
+        if(RobotConstants.DriverConstants.SQUARED_CONTROLLER_DRIVING) {
+            x = () -> Math.pow(driverXbox.getLeftX(), 2);
+            y = () -> Math.pow(driverXbox.getLeftY(), 2);
+            rot = () -> Math.pow(driverXbox.getRightX(), 2);
+        } else {
+            x = driverXbox::getLeftX;
+            y = driverXbox::getLeftY;
+            rot = driverXbox::getRightX;
+        }
```
Comment (Mboesn) [inline src/main/java/frc/robot/RobotContainer.java]: have the logic be inside the command, maybe have it get a boolean of SQUARED_CONTROLLER_DRIVING and a defualt set to DriverConstants.SQUARED_CONTROLLER_DRIVING

also import static

File: src/main/java/frc/robot/components/TrigonPlaystationController.java [hunk 4]
```diff
@@ -142,37 +142,23 @@ public void setRumble(double power) {
      */
     public void intermittentRumble(int quantity) {
         rumbleAmount = quantity * 2 - 1;
-        notifier.startPeriodic(kIntermittentRumbleTime);
+        notifier.startPeriodic(RobotConstants.DriverConstants.RUMBLE_INTERMISSION_TIME);
```
Comment (Mboesn) [inline src/main/java/frc/robot/components/TrigonPlaystationController.java]: import static

File: src/main/java/frc/robot/components/TrigonPlaystationController.java [hunk 5]
```diff
@@ -142,37 +142,23 @@ public void setRumble(double power) {
...
-     * @param hand Side of controller whose value should be returned.
-     * @return The X axis value of the controller.
-     */
-    @Override
-    public double getX(Hand hand) {
-        if(hand.equals(Hand.kLeft)) {
-            return getRawAxis(Axis.leftX.getAxis());
-        } else {
-            return getRawAxis(Axis.rightX.getAxis());
-        }
```
Comment (Mboesn) [inline src/main/java/frc/robot/components/TrigonPlaystationController.java:160]: why delete? both this and the new function can work

File: src/main/java/frc/robot/components/TrigonPlaystationController.java [hunk 6]
```diff
@@ -142,37 +142,23 @@ public void setRumble(double power) {
...
-     * @param hand Side of controller whose value should be returned.
-     * @return The Y axis value of the controller.
-     */
-    @Override
-    public double getY(Hand hand) {
-        if(hand.equals(Hand.kLeft)) {
-            return getRawAxis(Axis.leftY.getAxis());
-        } else {
-            return getRawAxis(Axis.rightY.getAxis());
-        }
```
Comment (Mboesn) [inline src/main/java/frc/robot/components/TrigonPlaystationController.java:175]: why delete? both this and the new function can work

File: src/main/java/frc/robot/components/TrigonXboxController.java [hunk 7]
```diff
@@ -21,86 +21,68 @@
...
 
         backButton = new JoystickButton(this, Button.kBack.value);
         startButton = new JoystickButton(this, Button.kStart.value);
         rumbleAmount = -1;
         notifier = new Notifier(this::notifierPeriodic);
     }
 
-    public JoystickButton getButtonA() {
-        return buttonA;
+    public JoystickButton getABtn() {
```
Comment (Mboesn) [inline src/main/java/frc/robot/components/TrigonXboxController.java:39]: Btn is not short for button

File: src/main/java/frc/robot/components/TrigonXboxController.java [hunk 8]
```diff
@@ -3,13 +3,13 @@
...
 public class TrigonXboxController extends XboxController {
-    private static final double kIntermittentRumbleTime = 0.15;
-    private final JoystickButton buttonA;
-    private final JoystickButton buttonB;
-    private final JoystickButton buttonX;
-    private final JoystickButton buttonY;
+    private final JoystickButton a;
+    private final JoystickButton b;
+    private final JoystickButton x;
+    private final JoystickButton y;
```
Comment (Mboesn) [inline src/main/java/frc/robot/components/TrigonXboxController.java]: i feel like the names with button is better
you don't call topMotor: top

File: src/main/java/frc/robot/components/TrigonXboxController.java [hunk 9]
```diff
@@ -21,86 +21,68 @@
...
     }
 
-    public JoystickButton getButtonY() {
-        return buttonY;
+    public JoystickButton getYBtn() {
+        return y;
     }
 
-    public JoystickButton getLeftBumper() {
+    public JoystickButton getLeftBumperBtn() {
```
Comment (Mboesn) [inline src/main/java/frc/robot/components/TrigonXboxController.java:55]: leftBumper not LeftBumperbutton

File: src/main/java/frc/robot/components/TrigonXboxController.java [hunk 10]
```diff
@@ -21,86 +21,68 @@
...
-     * @param positiveHand The hand that is positive, the other will be negative.
-     * @return positive trigger value - other trigger value.
-     */
-    public double getDeltaTriggers(Hand positiveHand) {
-
-        if(positiveHand == Hand.kRight)
-            return getTriggerAxis(Hand.kRight) - getTriggerAxis(Hand.kLeft);
-        else
-            return getTriggerAxis(Hand.kLeft) - getTriggerAxis(Hand.kRight);
-    }
```
Comment (Mboesn) [inline src/main/java/frc/robot/components/TrigonXboxController.java]: keep just change the logic to:
`return positiveHand == Hand.kRight ? getDeltaTriggers() : -getDeltaTriggers()`

File: src/main/java/frc/robot/components/TrigonXboxController.java [hunk 11]
```diff
@@ -120,21 +102,15 @@ public void setRumble(double power) {
      */
     public void intermittentRumble(int quantity) {
         rumbleAmount = quantity * 2 - 1;
-        notifier.startPeriodic(kIntermittentRumbleTime);
+        notifier.startPeriodic(RobotConstants.DriverConstants.RUMBLE_INTERMISSION_TIME);
```
Comment (Mboesn) [inline src/main/java/frc/robot/components/TrigonXboxController.java]: import static

File: src/main/java/frc/robot/components/TrigonXboxController.java [hunk 12]
```diff
@@ -120,21 +102,15 @@ public void setRumble(double power) {
...
     }
 
-    @Override
-    public double getX(Hand hand) {
-        if(Math.abs(super.getX(hand)) < 0.1)
-            return 0;
-        return super.getX(hand);
+    public double getLeftY() {
+        return -super.getLeftY();
     }
```
Comment (Mboesn) [inline src/main/java/frc/robot/components/TrigonXboxController.java:116]: why minus?

File: src/main/java/frc/robot/components/TrigonXboxController.java [hunk 13]
```diff
@@ -120,21 +102,15 @@ public void setRumble(double power) {
...
 
-    @Override
-    public double getY(Hand hand) {
-        if(Math.abs(super.getY(hand)) < 0.1)
-            return 0;
-        return -super.getY(hand);
+    public double getRightY() {
+        return -super.getRightY();
     }
```
Comment (Mboesn) [inline src/main/java/frc/robot/components/TrigonXboxController.java:111]: why minus?

File: src/main/java/frc/robot/components/TrigonXboxController.java [hunk 14]
```diff
@@ -21,86 +21,68 @@
...
+        x = new JoystickButton(this, Button.kX.value);
+        y = new JoystickButton(this, Button.kY.value);
+        leftBumper = new JoystickButton(this, Button.kLeftBumper.value);
+        rightBumper = new JoystickButton(this, Button.kRightBumper.value);
+        leftStickButton = new JoystickButton(this, Button.kLeftStick.value);
+        rightStickButton = new JoystickButton(this, Button.kRightStick.value);
 
         backButton = new JoystickButton(this, Button.kBack.value);
         startButton = new JoystickButton(this, Button.kStart.value);
         rumbleAmount = -1;
```
Comment (Mboesn) [inline src/main/java/frc/robot/components/TrigonXboxController.java:35]: magic number

File: src/main/java/frc/robot/subsystems/swerve/SwerveModule.java [hunk 15]
```diff
@@ -88,8 +88,8 @@ private static double placeInAppropriate0To360Scope(
         return newAngle;
     }
 
-    public int[] getEncoderVelocities() {
-        return new int[] {
+    public double[] getEncoderVelocities() {
+        return new double[] {
```
Comment (Mboesn) [inline src/main/java/frc/robot/subsystems/swerve/SwerveModule.java:92]: CTRE updated thier code? this compiles and works through simulation?

File: src/main/java/frc/robot/subsystems/swerve/SwerveSS.java [hunk 16]
```diff
@@ -62,7 +62,7 @@ public void drive(Translation2d translation, double rotation, boolean fieldRelat
                 )
         );
         // making sure the speeds are within the max speed
-        SwerveDriveKinematics.normalizeWheelSpeeds(swerveModuleStates, SwerveConstants.MAX_SPEED);
+        SwerveDriveKinematics.desaturateWheelSpeeds(swerveModuleStates, SwerveConstants.MAX_SPEED);
```
Comment (Mboesn) [inline src/main/java/frc/robot/subsystems/swerve/SwerveSS.java:65]: you sure this is the same function?

File: vendordeps/Phoenix-frc2022-latest.json [hunk 17]
```diff
@@ -0,0 +1,257 @@
+{
```
Comment (Mboesn) [inline vendordeps/Phoenix-frc2022-latest.json:1]: add rev library too

File: .gitignore [hunk 18]
```diff
@@ -160,3 +160,6 @@ imgui.ini
 
 # End of https://www.gitignore.io/api/c++,java,linux,macos,gradle,windows,visualstudiocode
 /.idea/
+simgui-ds.json
+simgui-window.json
+simgui.json
```
Comment (Mboesn) [inline .gitignore:165]: what is this?

**Review by Mboesn — CHANGES_REQUESTED** [review]
fix quick

File: build.gradle (hunk 2 above)
Comment (laviRZ) [reply build.gradle:66]: I used a brand new project's gradle.build file. I think this is fine

File: src/main/java/frc/robot/RobotContainer.java (hunk 3 above)
Comment (laviRZ) [reply src/main/java/frc/robot/RobotContainer.java]: I disagree. I think that the command is only getting a double supplier. If you want to square it, do it in the supplier.

File: src/main/java/frc/robot/components/TrigonPlaystationController.java (hunk 5 above)
Comment (laviRZ) [reply src/main/java/frc/robot/components/TrigonPlaystationController.java:160]: There's no `Hand` class anymore

File: src/main/java/frc/robot/components/TrigonPlaystationController.java (hunk 6 above)
Comment (laviRZ) [reply src/main/java/frc/robot/components/TrigonPlaystationController.java:175]: There's no `Hand` class anymore

File: src/main/java/frc/robot/components/TrigonXboxController.java (hunk 7 above)
Comment (laviRZ) [reply src/main/java/frc/robot/components/TrigonXboxController.java:39]: trust me, it is

File: src/main/java/frc/robot/components/TrigonXboxController.java (hunk 8 above)
Comment (laviRZ) [reply src/main/java/frc/robot/components/TrigonXboxController.java]: I think this is debatable.

File: src/main/java/frc/robot/components/TrigonXboxController.java (hunk 9 above)
Comment (laviRZ) [reply src/main/java/frc/robot/components/TrigonXboxController.java:55]: not possible, as there's a function with that name in the parent class of type `boolean`

File: src/main/java/frc/robot/components/TrigonXboxController.java (hunk 10 above)
Comment (laviRZ) [reply src/main/java/frc/robot/components/TrigonXboxController.java]: There's no `Hand` class anymore

File: src/main/java/frc/robot/components/TrigonXboxController.java (hunk 12 above)
Comment (laviRZ) [reply src/main/java/frc/robot/components/TrigonXboxController.java:116]: That's how it was before, isn't it?

File: src/main/java/frc/robot/components/TrigonXboxController.java (hunk 13 above)
Comment (laviRZ) [reply src/main/java/frc/robot/components/TrigonXboxController.java:111]: That's how it was before, isn't it?

File: src/main/java/frc/robot/components/TrigonXboxController.java (hunk 14 above)
Comment (laviRZ) [reply src/main/java/frc/robot/components/TrigonXboxController.java:35]: It's a counter

File: src/main/java/frc/robot/subsystems/swerve/SwerveModule.java (hunk 15 above)
Comment (laviRZ) [reply src/main/java/frc/robot/subsystems/swerve/SwerveModule.java:92]: yep

File: src/main/java/frc/robot/subsystems/swerve/SwerveSS.java (hunk 16 above)
Comment (laviRZ) [reply src/main/java/frc/robot/subsystems/swerve/SwerveSS.java:65]: image

File: .gitignore (hunk 18 above)
Comment (laviRZ) [reply .gitignore:165]: look at commit name

File: src/main/java/frc/robot/RobotContainer.java (hunk 3 above)
Comment (Mboesn) [reply src/main/java/frc/robot/RobotContainer.java]: but this is way too long

File: src/main/java/frc/robot/components/TrigonXboxController.java [hunk 19]
```diff
@@ -120,21 +102,15 @@ public void setRumble(double power) {
...
-        notifier.startPeriodic(kIntermittentRumbleTime);
+        notifier.startPeriodic(RobotConstants.DriverConstants.RUMBLE_INTERMISSION_TIME);
     }
 
-    @Override
-    public double getY(Hand hand) {
-        if(Math.abs(super.getY(hand)) < 0.1)
-            return 0;
-        return -super.getY(hand);
+    public double getRightY() {
```
Comment (Mboesn) [inline src/main/java/frc/robot/components/TrigonXboxController.java:109]: add @ override

File: src/main/java/frc/robot/components/TrigonXboxController.java [hunk 20]
```diff
@@ -120,21 +102,15 @@ public void setRumble(double power) {
...
+        return -super.getRightY();
     }
 
-    @Override
-    public double getX(Hand hand) {
-        if(Math.abs(super.getX(hand)) < 0.1)
-            return 0;
-        return super.getX(hand);
+    public double getLeftY() {
```
Comment (Mboesn) [inline src/main/java/frc/robot/components/TrigonXboxController.java:114]: add @ override

File: src/main/java/frc/robot/components/TrigonXboxController.java (hunk 10 above)
Comment (Mboesn) [reply src/main/java/frc/robot/components/TrigonXboxController.java]: than find a boolean instead

**Review by Mboesn — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/components/TrigonXboxController.java (hunk 7 above)
Comment (Mboesn) [reply src/main/java/frc/robot/components/TrigonXboxController.java:39]: no

File: src/main/java/frc/robot/components/TrigonXboxController.java (hunk 10 above)
Comment (laviRZ) [reply src/main/java/frc/robot/components/TrigonXboxController.java]: boolean? that's ugly. Just use the default. worst case just multiply this by -1.

File: src/main/java/frc/robot/components/TrigonXboxController.java (hunk 8 above)
Comment (Mboesn) [reply src/main/java/frc/robot/components/TrigonXboxController.java]: and I say it's better as buttonA

File: src/main/java/frc/robot/components/TrigonXboxController.java (hunk 7 above)
Comment (laviRZ) [reply src/main/java/frc/robot/components/TrigonXboxController.java:39]: image

File: src/main/java/frc/robot/components/TrigonXboxController.java (hunk 8 above)
Comment (laviRZ) [reply src/main/java/frc/robot/components/TrigonXboxController.java]: aBtn if you want to add it to the name

File: src/main/java/frc/robot/components/TrigonXboxController.java (hunk 7 above)
Comment (Mboesn) [reply src/main/java/frc/robot/components/TrigonXboxController.java:39]: but it's disgusting

File: src/main/java/frc/robot/components/TrigonXboxController.java (hunk 8 above)
Comment (Mboesn) [reply src/main/java/frc/robot/components/TrigonXboxController.java]: sure

File: src/main/java/frc/robot/components/TrigonXboxController.java (hunk 7 above)
Comment (laviRZ) [reply src/main/java/frc/robot/components/TrigonXboxController.java:39]: btn is not

File: src/main/java/frc/robot/RobotContainer.java [hunk 21]
```diff
@@ -40,23 +40,14 @@ private void initializeSubsystems() {
...
         driveWithXboxCMD = new SupplierDriveCMD(
                 swerveSS,
-                x,
-                y,
-                rot,
+                squared ? ()->Math.pow(driverXbox.getLeftX(), 2) : driverXbox::getLeftX,
+                squared ? ()->Math.pow(driverXbox.getLeftY(), 2) : driverXbox::getLeftY,
+                squared ? ()->Math.pow(driverXbox.getRightX(), 2) : driverXbox::getRightX,
                 true);
+
```
Comment (Mboesn) [inline src/main/java/frc/robot/RobotContainer.java:53]: blank line

**Review by Mboesn — CHANGES_REQUESTED** [review]
commit, check if it simulates then merge

File: src/main/java/frc/robot/RobotContainer.java (hunk 21 above)
Comment (laviRZ) [reply src/main/java/frc/robot/RobotContainer.java:53]: It's there on purpose
Don't we want to set all of the default commands at the end of the function, separate from the initialization?

Comment (Mboesn) [reply src/main/java/frc/robot/RobotContainer.java:53]: add bind commands

**Review by Mboesn — APPROVED** [review]

**Pull request merged**